### PR TITLE
Force OPTIONS caching, remove unused DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE build flag

### PR DESCRIPTION
**Pull request by jschr**
This PR allows OPTIONS requests to be cacheable. Previously if the network went down, any OPTIONS requests would fail to receive a response, causing all CORS requests to fail as well. 

I traced this down to the DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE build flag, which we are setting to `1` by default. When this flag is enabled we remove this file here: https://github.com/mirainc/funes/blob/master/conf/forward_cache_expiry_override/expiry.conf

By removing this file we are no longer telling nginx to cache the OPTIONS request here: https://github.com/mirainc/funes/blob/master/conf/nginx.conf.server.template#L54.

I'm not sure why we are setting `DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE=1` by default it's been like this for several years. 

Instead of enabling it to fix I instead removed this feature since we don't seem to be using it and added an explicit config to cache the OPTIONS request only.

### CODE REVIEW

File: conf/nginx.conf.server.template [hunk 1]
```diff
@@ -18,41 +18,12 @@ resolver ${NAMESERVER} ipv6=off valid=10s;
...
-    default                     $content_type_expiry;
-    # foo.com                   30d;                        # example of expiry by host
-    "~*uplynk.com"              -1s;                        # known HLS stream provider
-}
 
-map $uri $uri_expiry {
-    default                     $host_expiry;
-    # "~*.gifv"                 30d;                        # example of expiry by uri/file extension
+map $request_method $cache_override_expiry {
+    'OPTIONS'                   1s;
```
Comment (jschr) [inline conf/nginx.conf.server.template:23]: My fix here is to ensure we set the expiry of OPTIONS header to a low value like 1s. This is what allows nginx to cache the OPTIONS request, which can be used when the device is offline. 

I've set this to a low value to avoid impacting any web applications while the device is online. When offline, the OPTIONS request will continue to pull from cache (even after 1s).

File: conf/nginx.conf.server.template [hunk 2]
```diff
@@ -18,41 +18,12 @@ resolver ${NAMESERVER} ipv6=off valid=10s;
...
-    'video/mp4'                             max;
-    'application/binary'                    -1s;            # used by HLS streams
-    '~*mpegurl'                             -1s;
-}
-
-map $host $host_expiry {
-    default                     $content_type_expiry;
-    # foo.com                   30d;                        # example of expiry by host
-    "~*uplynk.com"              -1s;                        # known HLS stream provider
-}
```
Comment (jschr) [inline conf/nginx.conf.server.template:44]: These settings were never actually used because we were setting DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE=1, which removes the `expires $cache_expiry;` part of the config. I've removed them to avoid confusion in the future.

File: conf/forward_cache_expiry_override/expiry.conf
Comment (jschr) [inline conf/forward_cache_expiry_override/expiry.conf:1]: When DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE=1 is set, this file is removed when building here: https://github.com/mirainc/funes/blob/55c4ab7907c69c6de16c0a4a5053691423e5072b/config.sh#L46 

When this flag is set, the `$cache_expiry` variable is never actually used.

**Comment by jschr** [comment]
> @jschr I think you will need to also update https://github.com/mirainc/screenray/blob/main/funes/Dockerfile#L36

I've got a draft PR I'm testing with: https://github.com/mirainc/screenray/pull/933

**Comment by jschr** [comment]
> I don't understand exactly what's happening here :/ but it sounds like you've tested the specific case so I'm happy to approve.

Sorry it was a bit hard to explain 😅 but happy to chat about it for a more thorough overview. 

TLDR; OPTIONS request were not being cached because the DISABLE_FORWARD_CACHE_EXPIRY_OVERRIDE=1 told it not to. But I didn't want to set it to `0` because it would bring in other config changes that haven't been used in years so I removed that flag and instead only added the necessary config to cache the OPTIONS request.

File: conf/nginx.conf.server.template (hunk 1 above)
Comment (jschr) [inline conf/nginx.conf.server.template:23]: TODO: Would be helpful to add a comment explaining why this is here.
